src/schema.rs

```diff
@@ -421,10 +409,87 @@ impl Schema {
         }
     }
 
-    /// Parse a `serde_json::Value` representing a primitive Avro type into a
-    /// `Schema`.
-    fn parse_primitive(primitive: &str) -> AvroResult<Self> {
-        match primitive {
+    /// Create a `Schema` from a string representing a JSON Avro schema.
+    pub fn parse_str(input: &str) -> Result<Schema, Error> {
+        let mut parser = Parser::default();
+        parser.parse_str(input)
+    }
+
+    /// Create a array of `Schema`'s from a list of JSON Avro schemas. It is allowed that
+    /// the schemas have cross-dependencies; these will be resolved during parsing.
+    pub fn parse_list(input: &[&str]) -> Result<Vec<Schema>, Error> {
+        let mut input_schemas: HashMap<String, Value> = HashMap::with_capacity(input.len());
+        for js in input {
+            let schema: Value = serde_json::from_str(js).map_err(Error::ParseSchemaJson)?;
+            if let Value::Object(inner) = &schema {
+                let fullname = Name::parse(&inner)?.fullname(None);
+                let _ = input_schemas.insert(fullname, schema);
+            } else {
+                return Err(Error::GetNameField);
+            }
+        }
+        let mut parser = Parser {
+            input_schemas,
+            parsed_schemas: HashMap::with_capacity(input.len()),
+        };
+        parser.parse_list()
+    }
+
+    pub fn parse(value: &Value) -> AvroResult<Schema> {
+        let mut parser = Parser::default();
+        parser.parse(value)
+    }
+}
+
+impl Parser {
+    /// Create a `Schema` from a string representing a JSON Avro schema.
+    fn parse_str(&mut self, input: &str) -> Result<Schema, Error> {
+        // TODO: (#82) this should be a ParseSchemaError wrapping the JSON error
+        let value = serde_json::from_str(input).map_err(Error::ParseSchemaJson)?;
+        self.parse(&value)
+    }
+
+    /// Create a array of `Schema`'s from a an iterator of JSON Avro schemas. It is allowed that
+    /// the schemas have cross-dependencies; these will be resolved during parsing.
+    fn parse_list(&mut self) -> Result<Vec<Schema>, Error> {
+        while !self.input_schemas.is_empty() {
+            let next_name = self
+                .input_schemas
+                .keys()
+                .next()
+                .expect("Input schemas unexpectedly empty")
+                .to_owned();
+            let (name, value) = self
+                .input_schemas
+                .remove_entry(&next_name)
+                .expect("Key unexpectedly missing");
+            let parsed = self.parse(&value)?;
+            self.parsed_schemas.insert(name, parsed);
+        }
+
+        let mut parsed_schemas = Vec::new();
```

Comment:
nit: can we use `Vec::with_capacity(self.parsed_schemas.len())` here?

Reply:
:+1: